test/test_async_closures.py

```diff
@@ -0,0 +1,93 @@
+"""Tests for xla_dist."""
+from __future__ import division
+from __future__ import print_function
+from threading import Event
+from time import sleep
+
+import unittest
+from unittest import mock
+
+import torch
+import torch_xla.core.xla_model as xm
+
+
+class AsyncClosuresTest(unittest.TestCase):
+
+  def test_synchronous(self):
+    flag = Event()
+    assert not flag.is_set()
+
+    def closure():
+      sleep(1)
+      assert not flag.is_set()
+      flag.set()
+
+    xm.add_step_closure(closure)
+    xm.mark_step()
+
+    # should not get to this part before closure is finished running
+    assert flag.is_set()
+
+  def test_asynchronous(self):
+    flag = Event()
+    assert not flag.is_set()
+
+    def closure():
+      sleep(1)
+      assert flag.is_set()
+
+    xm.add_step_closure(closure, run_async=True)
+    xm.mark_step()
+
+    # should get to this part and complete before closure is finished running
+    assert not flag.is_set()
+    flag.set()
+
+  def test_synchronous_exception(self):
+    flag = Event()
+    assert not flag.is_set()
+
+    try:
+
```

Comment:
Is empty line here added by yapf? If not can we remove them?

Reply:
Yes, that was added by yapf.